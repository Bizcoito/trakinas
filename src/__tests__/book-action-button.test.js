jest.mock('../firebase-manager');

import React from 'react';
import BookActionButton from '../components/book-action-button';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('BookActionButton', () => {
  const book = {
    bookId: '1235asdf',
    name: 'ASDF',
    description: 'SistemasDeInformação',
    thumbnail: 'http://pix.toile-libre.org/upload/original/1499647888.png',
    available: true,
    xunda: 'xunda',
  }

  const bookLiteralStates = {
    available: 'Borrow book',
    rented: 'Return book',
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<BookActionButton book={book} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows an available book', () => {
    const component = mount(<BookActionButton book={book} />);
    expect(component.state().book.available).toEqual(book.available);
    expect(component.find('button').text()).toEqual(bookLiteralStates.available);
  });

  it('changes book available and component state after click', () => {
    const component = mount(<BookActionButton book={book} />);
    const expectedAvailableStateAfterClick = !book.available;

    component.find('button').simulate('click');

    expect(component.state().book.available).toEqual(expectedAvailableStateAfterClick);
  });
});
