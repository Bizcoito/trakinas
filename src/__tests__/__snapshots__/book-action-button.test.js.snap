// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookActionButton renders correctly 1`] = `
<button
  className="btn btn-info"
  onClick={[Function]}
>
  Borrow book
</button>
`;
